pkg/manager-server/server_local.go

```diff
@@ -177,52 +162,63 @@ func (c *LocalServerController) discoverUsingSuppliedDevices(s *Storage, devices
 	return nil
 }
 
-func (c *LocalServerController) discoverViaNamespaces(s *Storage) error {
-	paths, err := c.namespaces()
+// Discover NVMe Namespaces
+func (c *LocalServerController) discoverViaNamespaces() error {
+
+	data, err := c.run("nvme list --output-format=json")
 	if err != nil {
 		return err
 	}
 
-	for _, path := range paths {
-
-		sns, err := c.newStorageNamespace(path)
-		if err != nil {
-			return err
-		}
+	devices := nvmeListDevices{}
+	if err := json.Unmarshal(data, &devices); err != nil {
+		return err
+	}
 
-		discovered := false
-		for _, ns := range s.discoveredNamespaces {
-			if ns.Equals(sns) {
-				discovered = true
-				break
-			}
-		}
+	for _, storage := range c.storages {
 
-		if discovered {
+		// Skip storages that already discovered the expected namespaces
+		if len(storage.expectedNamespaces) == len(storage.discoveredNamespaces) {
```

Comment:
When a server storage resource gets created, it comes with a list of expected namespaces. If they are already found, there's no point in rediscovering them. You could compare the lists, but you know they are identical from the length alone because the discovery code below ensures there are no duplicates.